### ISSUE TITLE
easy way to create multi-level feature assertions

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#87]
*Platform* (JVM, JS and/or Android): 

## Code related feature
Often when I come across code bases I see something along the lines of
```
expect(person).isA<Student>{
  expect(subject.universities.first().name).toBe("Blabla")
  expect(subject.name).toBe("robert)
}
```
This works but is not ideal because:
- the reporting will neither include information about `person` nor `universities`
- the assertion group block does not evaluate both "feature-assertions" as they are effectively not nested assertions where all of them throw if they fail

I think we could enhance Atrium, guide the user in the right direction:
- :heavy_check_mark: provide an easier way to create multi-level feature assertions (the above could be rewritten using `property(subject::universities).returnValueOf(subject::first).property(subject::name)` with the benefit that everything is included in reporting but I guess enough people don't care about it. Thus something along the line of changeSubject but with a text something like `quick feature: value` in reporting would be useful => we have `feature("text") { foo().bar }` which is good enough for now. We will improve when https://github.com/christophsturm/filepeek works for multiplatform
- decide if we should add `expect`, `assert` and `assertThat` as extension function of `Assert` so that they act as alias of the function above => this way they do not break the assertion block semantic and `person` would be included in the error reporting
- decide whether we deprecate the functions and point the user in the correct direction (I guess that wouldn't be too bad)

**Comment by robstoll** [comment]
Now that subject is hidden, one cannot longer use `expect(subject...)` within a sub assertion. 
We need to guide the user to use feature assertions instead

**Comment by robstoll** [comment]
part 2 implemented by db14c7a1
